startup/gui/layouts.py

```diff
@@ -49,9 +49,81 @@
+def createDefault( scriptNode ):
+	GafferUI.CompoundEditor(
+		scriptNode, children =
+		(
+			GafferUI.SplitContainer.Orientation.Vertical, 0.97,
+			(
+				(
+					GafferUI.SplitContainer.Orientation.Horizontal, 0.70,
+					(
+						(
+							GafferUI.SplitContainer.Orientation.Vertical, 0.48,
+							(
+								{ 'tabs' : (GafferUI.Viewer( scriptNode ),  ), 'tabsVisible' : True, 'currentTab' : 0 },
+								{ 'tabs' : (GafferUI.NodeGraph( scriptNode ), GafferSceneUI.PrimitiveInspector( scriptNode ),), 'tabsVisible' : True, 'currentTab' : 0 }
+							)
+						),
+						(
+							GafferUI.SplitContainer.Orientation.Vertical, 0.54,
+							(
+								{ 'tabs' : (GafferUI.NodeEditor( scriptNode ), GafferSceneUI.SceneInspector( scriptNode )), 'tabsVisible' : True, 'currentTab' : 0 },
+								{ 'tabs' : (GafferSceneUI.HierarchyView( scriptNode ), GafferUI.ScriptEditor( scriptNode )), 'tabsVisible' : True, 'currentTab' : 0 }
+							)
+						)
+					)
+				),
+				{ 'tabs' : (GafferUI.Timeline( scriptNode ),), 'tabsVisible' : False, 'currentTab' : 0 }
+			)
+		)
+	)
+
+def createScene( scriptNode ):
+	GafferUI.CompoundEditor(
+		scriptNode, children =
+		(
+			GafferUI.SplitContainer.Orientation.Horizontal, 0.772395,
+			(
+				(
+					GafferUI.SplitContainer.Orientation.Horizontal, 0.256257,
+					(
+						{'tabs': (GafferSceneUI.HierarchyView( scriptNode ),), 'tabsVisible': True, 'currentTab': 0, 'pinned': [False]},
+						(
+							GafferUI.SplitContainer.Orientation.Vertical, 0.500620,
+							(
+								(
+									GafferUI.SplitContainer.Orientation.Vertical, 0.929648,
+									(
+										{'tabs': (GafferUI.Viewer( scriptNode ),), 'tabsVisible': True, 'currentTab': 0, 'pinned': [False]},
+										{'tabs': (GafferUI.Timeline( scriptNode ),), 'tabsVisible': False, 'currentTab': 0, 'pinned': [None]}
+									)
+								),
+								{'tabs': (GafferUI.NodeGraph( scriptNode ),), 'tabsVisible': True, 'currentTab': 0, 'pinned': [None]}
+							)
+						)
+					)
+				),
+				(
+					GafferUI.SplitContainer.Orientation.Vertical, 0.500620,
+					(
+						{'tabs': (GafferUI.NodeEditor( scriptNode ),), 'tabsVisible': True, 'currentTab': 0, 'pinned': [False]},
+						{'tabs': (GafferSceneUI.SceneInspector( scriptNode ),), 'tabsVisible': True, 'currentTab': 0, 'pinned': [False]}
+					)
+				)
+			)
+		)
+	)
+
+def getSource(fn):
```

Comment:
I don't think defining an unused function only to suck the source out of it is an improvement. The reformatting is an improvement if we want to edit these layouts by hand, but I think what I've actually been doing in practice is modifying them in the UI and then saving them. Let's either revert it all, or keep the formatting but in a simple triple-quoted string.

Reply:
> The reformatting is an improvement if we want to edit these layouts by hand, but I think what I've actually been doing in practice is modifying them in the UI and then saving them.

Ah! This explains the strange situation of python code in the python file as a mushed together in a string. I'll revert and add a few comments for future visitors to this file.